packages/kbn-config-schema/src/types/object_type.test.ts

```diff
@@ -30,13 +30,20 @@ test('returns value by default', () => {
   expect(type.validate(value)).toEqual({ name: 'test' });
 });
 
+test('returns empty object if undefined', () => {
+  const type = schema.object({});
+  expect(type.validate(undefined)).toEqual({});
```

Comment:
as of me it's weird behaviour. other types throw in this case https://github.com/elastic/kibana/blob/7949f34e0c64a90f4eb1879d1c61d116582ffd75/packages/kbn-config-schema/src/types/string_type.test.ts#L31

Reply:
I think for primitive / non-composite types that makes sense, but in the case 
of composite types like Object, we almost always want the defaults for each key to be applied, especially for the primary use case, config.

The only other option I can think of is allowing a `defaultValue: {}` on schema.object that would still allow the defaults to be applied to subkeys?

Reply:
The behavior you are expecting does happen in the `applyDefaults: false` case.

Reply:
> I think for primitive / non-composite types that makes sense, but in the case
of composite types like Object,

but it's not the case for array, record. 
```js
  const type = schema.arrayOf(schema.string());
  expect(() => type.validate(undefined)).toThrowErrorMatchingInlineSnapshot(
    `"expected value of type [array] but got [undefined]"`
  );
```
but okay